### PR TITLE
July Sprint work

### PR DESCRIPTION
**Pull request by pdl**
Fixes for:

- https://unep-wcmc.codebasehq.com/projects/cites-support-maintenance/tickets/240
- https://unep-wcmc.codebasehq.com/projects/cites-support-maintenance/tickets/237
- https://unep-wcmc.codebasehq.com/projects/cites-support-maintenance/tickets/266
- https://unep-wcmc.codebasehq.com/projects/cites-support-maintenance/tickets/220

Also extra validation called for by https://unep-wcmc.codebasehq.com/projects/cites-support-maintenance/tickets/237

### CODE REVIEW

File: db/migrate/20240723164647_update_conversion_rules_for_trade_plus_formatted_data_view.rb
Comment (leonardow-unep-wcmc) [inline db/migrate/20240723164647_update_conversion_rules_for_trade_plus_formatted_data_view.rb:1]: Why up and down using the same version SQL file?

File: db/migrate/20240725145653_create_trade_group_and_rule_tables.rb
Comment (leonardow-unep-wcmc) [inline db/migrate/20240725145653_create_trade_group_and_rule_tables.rb:1]: This file may not work, lets have a call

File: db/migrate/20240729120000_replace_trade_plus_views.rb
Comment (leonardow-unep-wcmc) [inline db/migrate/20240729120000_replace_trade_plus_views.rb:1]: Why up and down using the same version SQL file?

File: app/models/common_name.rb [hunk 1]
```diff
@@ -21,6 +21,15 @@ class CommonName < ApplicationRecord
   validates :name, :presence => true,
     :uniqueness => { :scope => :language_id }
 
+  validate :enforce_latin_chars_for_pdf
+
+  def enforce_latin_chars_for_pdf
```
[the remark this answers is not in the document]
Comment (pdl) [reply app/models/common_name.rb]: Done

File: lib/tasks/import_trade_rules.rake [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+    path_to_rules_file = "#{Rails.application.root}/lib/data/trade_mapping.yml"
+    abort("File doesn't exist.") unless File.exists?(path_to_rules_file)
+    rules_definition = YAML.load_file(path_to_rules_file)
+
+    ApplicationRecord.transaction do
+      Trade::ConversionRule.delete_all
+      Trade::TaxonGroup.delete_all # skips AR triggers
+
+      groups_definition.each do |row|
+        Trade::TaxonGroup.create!(row)
```
[the remark this answers is not in the document]
Comment (pdl) [reply lib/tasks/import_trade_rules.rake:19]: Yeah, these rules are manually curated and updated once every three years, so the file will always be small.

File: db/migrate/20240723164647_update_conversion_rules_for_trade_plus_formatted_data_view.rb
Comment (pdl) [reply db/migrate/20240723164647_update_conversion_rules_for_trade_plus_formatted_data_view.rb:1]: Some sql files are the same, some are not.

File: db/migrate/20240725145653_create_trade_group_and_rule_tables.rb
Comment (pdl) [reply db/migrate/20240725145653_create_trade_group_and_rule_tables.rb:1]: It does work - question was about whether you could create views on tables in the same transaction.

File: db/migrate/20240729120000_replace_trade_plus_views.rb
Comment (pdl) [reply db/migrate/20240729120000_replace_trade_plus_views.rb:1]: Some views use same, others are different.

File: spec/models/taxon_concept/validation_spec.rb [hunk 3]
```diff
@@ -180,5 +180,16 @@
...
     end
+    context "author_year" do
+      specify "is valid with lots of non-ASCII PDF-safe characters" do
+        species.author_year = 'Sigríður O’Brian–Żądło (2003)'
+        expect(species.error_on(:author_year).size).to eq(0)
+      end
+
+      specify "is not valid with cyrillic characters" do
+        species.author_year = 'Дмитровна (2001)'
+        expect(species.error_on(:author_year).size).to eq(1)
```
[the remark this answers is not in the document]
Comment (pdl) [reply spec/models/taxon_concept/validation_spec.rb]: Done

File: .rubocop.yml
[the remark this answers is not in the document]
Comment (pdl) [reply .rubocop.yml:1]: Yeah. These additions were taken from ORS plus some tweaks I made in Qatar. I've not applied this to the whole codebase, so I'll remove it from the PR and do that in another PR.

**Pull request merged**